src/app/ui/editor/select_box_state.cpp

```diff
@@ -250,47 +289,225 @@ void SelectBoxState::postRenderDecorator(EditorPostRender* render)
   const gfx::Color gridColor = skin::SkinTheme::instance()->colors.selectBoxGrid();
   const gfx::Point mainOffset = editor->mainTilePosition();
   gfx::Rect rc = getBoxBounds();
+  gfx::Size padding;
+  if (hasFlag(Flags::PaddingRulers)) {
+    padding = getPaddingBounds();
+    if (padding.w < 0)
```

Comment:
I think these `padding.w < 0` checks should be in `getPaddingBounds()`, so you can move 
```
   if (padding.w < 0)
      padding.w = 0;
    if (padding.h < 0)
      padding.h = 0;
```
inside `getPaddingBounds()` before calling `gfx::Size(w, h)` constructor with negative `w` and `h` values (side note: indeed, a `gfx::Size()` should be a `gfx::SizeT<unsigned int>`, to avoid using negative numbers, but that is a different history for other time).